### PR TITLE
Add missing lock around vInventoryToSend vector in respend relayer unit test

### PR DESCRIPTION
**Pull request by sickpig**

### CODE REVIEW

File: src/net.h [hunk 1]
```diff
@@ -762,6 +762,12 @@ class CNode
         vInventoryToSend.push_back(inv);
     }
```
Comment (gandrewstone) [inline src/net.h:770]: please add doxygen comment (and don't hold back if you want to also doc a few of the functions near your addition!)

Comment (sickpig) [reply src/net.h:770]: ok will do

Comment (sickpig) [reply src/net.h:770]: done

File: src/net.h [hunk 2]
```diff
@@ -762,12 +768,18 @@ class CNode
...
 
+    /** Get size onf INVs inventory n a thread safe way*/
     unsigned int GetInventoryToSendSize()
```
Comment (ptschip) [inline src/net.h:772]: -inventory n
+inventory in

**Pull request merged**